scrape_books/spiders/books.py

```diff
@@ -0,0 +1,49 @@
+import re
+
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/catalogue/category/books_1/page-1.html"]
+
+    def parse(self, response: Response, **kwargs):
+        for book in response.css(".product_pod"):
+            book_url = response.urljoin(book.css("h3 > a::attr(href)").get())
+            yield scrapy.Request(book_url, callback=self._parse_book_from_detail_page)
+
+        next_page = response.css("ul > li.next > a::attr(href)").get()
+        if next_page:
+            next_page = response.urljoin(next_page)
+            yield scrapy.Request(next_page, callback=self.parse)
+
+    def _parse_book_from_detail_page(self, response: Response) -> dict:
+        title = response.css("h1::text").get()
+        price = float(response.css("p.price_color::text").get().replace("£", ""))
```

Comment:
Ensure that the price string is not None before attempting to replace and convert it to a float. Consider adding a check or default value to handle cases where the price might not be found.